cmd/virt-chroot/main.go

```diff
@@ -57,50 +57,14 @@ func main() {
 				}
 			}
 
-			if cpuTime > 0 {
-				value := &syscall.Rlimit{
-					Cur: cpuTime,
-					Max: cpuTime,
-				}
-				err := syscall.Setrlimit(unix.RLIMIT_CPU, value)
-				if err != nil {
-					return fmt.Errorf("error setting prlimit on cpu time with value %d: %v", value, err)
-				}
-			}
-
-			if memoryBytes > 0 {
-				value := &syscall.Rlimit{
-					Cur: memoryBytes,
-					Max: memoryBytes,
-				}
-				err := syscall.Setrlimit(unix.RLIMIT_AS, value)
-				if err != nil {
-					return fmt.Errorf("error setting prlimit on virtual memory with value %d: %v", value, err)
-				}
+			resourceLimitErr := setResourceRLimits(cpuTime, memoryBytes)
```

Comment:
Thanks for proposing to reduce cognitive load. Whenever I see a comment I think: hmm, this should probably sit in its own function.

The calling function does not differentiate between `resourceLimitErr` or `userErr`; To reduce cognitive load, I'd call both of them `err`.